test/distributed/test_control_collectives.py

```diff
@@ -0,0 +1,159 @@
+# Owner(s): ["oncall: distributed"]
+
+from datetime import timedelta
+from multiprocessing.pool import ThreadPool
+
+import torch
+import torch.distributed as dist
+from torch.testing._internal.common_utils import run_tests, TestCase
+
+
+class TestCollectives(TestCase):
+    def test_barrier(self) -> None:
+        store = dist.HashStore()
+
+        world_size = 2
+
+        def f(rank: int) -> None:
+            collectives = dist.StoreCollectives(store, rank, world_size)
+            collectives.barrier("foo", timedelta(seconds=10), True)
```

Comment:
Yeah, I mean this 'foo' does not really provide much value for users? could be optional?

Reply:
For store, this needs to be unique, if you call it multiple times with "foo" it won't actually do any barrier operation

Reply:
OK, I was thinking it would be much nicer and cleaner if our general API could remove the burden from users of maintaining an unique key for each collective due to implementation limitations. Can we maintain a seqID or something within the implementation?

Reply:
we could potentially -- it gets a bit risky when you consider the case that not all workers may be joining at the same time. In elastic it's elastic so a simple seqID may not be sufficient and you need to key it off of the rendezvous round instead

Reply:
OK, this makes sense in elastic